docs/tutorial/helper_scripts.md

```diff
@@ -0,0 +1,176 @@
+
+## colliderinit
+
+This is used to convert data `D_address` in the various ColliderInit functions into the format of a collider. It lives in `tools/overlayhelpers`. Because there are different types of collider, you need to give it the type of collider as well. This does not need the baserom path, and a recent update allows it to be run from anywhere. You also have to give it the `<address>` without the leading `D_`.
+```sh
+./colliderinit.py <address> <type> <num>
+```
+Collider types supported are
+
+- `ColliderJntSphInit`
+- `ColliderCylinderInit`
+- `ColliderTrisInit`
+- `ColliderQuadInit`
+- `ColliderJntSphElementInit`
+- `ColliderTrisElementInit`
+
+and `num` is used only for `ColliderJntSphElementInit`.
+
+## sfxconvert
+
+Automatically converts sound effect numbers in a file into their corresponding `#defines`, taking into account if `SFX_FLAG` is used. Run on a specific C file,
+```sh
+./tools/sfxconvert.py <path_to_file> <path_to_repo>
+```
+
+Optional arguments are `-o output` to output to a different file and `-v` to give verbose output (i.e. tell you what changes it has made).
+
+## vtfmt
+
+This turns the strange strings in the `osSyncPrintf`s into the human-readable equivalent instructions. Copy the contents, including the quotation marks, and run
+```sh
+./tools/vtfmt.py "contents"
+```
+and replace the contents of the printf with the output.
+
+## Glank's N64 tools
+
+In particular, the ones used to decompile graphics macros. Their use is discussed in the section on [decompiling Draw functions](draw_functions.md).
+
+## graphovl
+
+This generates a directed graph showing an actor's function. Search for `graphovl.py` in the Discord. Put it in the root directory of the project, and run
+```sh
+./graphovl.py Actor_Name
+```
+to produce a png in the `graphs` subdirectory.
+
+## format
+
+Shell script that does a standardised format to the C code. Can be run on a file, a directory, or the whole codebase. Run this before you submit a PR.
+
+## find_unused_asm
+
+Tracks down any `.s` files no longer used by the project. Does not ignore comments, so you have to actually remove any `#pragma` lines for it to consider the file unused.
+```sh
+./tools/find_unused_asm.py
+```
+will output a list of all such files, while adding `-d` deletes the files.
+
+## csdis
+
+This converts the cutscene data into macros that the cutscene system uses. Cutscenes are generally very long, so I recommend sending the output straight to a file with `>`, rather than trying to copy it all from the terminal. Run
+```sh
+./tools/csdis.py <address>
+```
+on the address from the `D_address` containing the cutscene data.
+
+## regconvert
+
+This converts the direct memory references, of the form `gGameInfo + 0x<offset>`, into the corresponding REG macros defined in [regs.h](../include/regs.h). Run
```

Comment:
Currently, this script will works with `gGameInfo->data` indexes by default. For offsets, you need to pass the flag `--offset`. It may be worth noting the flag `-file` which replaces automatically each occurrence of that array in a file.